SEDaily-IOS/Podcast.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  Podcast.swift
+//  SEDaily-IOS
+//
+//  Created by Craig Holliday on 10/12/17.
+//  Copyright © 2017 Koala Tea. All rights reserved.
+//
+
+import Foundation
+
+enum PodcastTypes: String {
+    case new = "new"
+    case top = "top"
+    case recommended = "recommended"
+}
+
+enum PodcastCategoryIds: Int {
+    case All = -1
+    case Business_and_Philosophy = 1068
+    case Blockchain = 1082
+    case Cloud_Engineering = 1079
+    case Data = 1081
+    case JavaScript = 1084
+    case Machine_Learning = 1080
+    case Open_Source = 1078
+    case Security = 1083
+    case Hackers = 1085
+    case Greatest_Hits = 1069
+    
+    var description: String {
+        switch self {
+        case .All:
+            return L10n.tabTitleAll
+        case .Business_and_Philosophy:
+            return L10n.tabTitleBusinessAndPhilosophy
+        case .Blockchain:
+            return L10n.tabTitleBlockchain
+        case .Cloud_Engineering:
+            return L10n.tabTitleCloudEngineering
+        case .Data:
+            return L10n.tabTitleData
+        case .JavaScript:
+            return L10n.tabTitleJavaScript
+        case .Machine_Learning:
+            return L10n.tabTitleMachineLearning
+        case .Open_Source:
+            return L10n.tabTitleOpenSource
+        case .Security:
+            return L10n.tabTitleSecurity
+        case .Hackers:
+            return L10n.tabTitleHackers
+        case .Greatest_Hits:
+            return L10n.tabTitleGreatestHits
+        }
+    }
+}
+
+public struct Podcast: Codable {
+    let _id: String
+    let date: String
+    let link: String
+    let categories: [Int]?
+    let tags: [Int]?
+    let mp3: String
+    let featuredImage: String?
+    struct Content: Codable {
+        let rendered: String
+    }
+    let content: Content
+    struct Title: Codable {
+        let rendered: String
+    }
+    let title: Title
+    let score: Int
+    var type: String? = "new"
+}
+
+extension Podcast: Equatable {
+    public static func ==(lhs: Podcast, rhs: Podcast) -> Bool {
+        return lhs._id == rhs._id &&
+            lhs.date == rhs.date &&
+            lhs.link == rhs.link &&
+            lhs.categories ?? [] == rhs.categories ?? [] &&
+            lhs.tags ?? [] == rhs.tags ?? [] &&
+            lhs.mp3 == rhs.mp3 &&
+            lhs.featuredImage == rhs.featuredImage &&
+            lhs.content.rendered == rhs.content.rendered &&
+            lhs.title.rendered == rhs.title.rendered &&
+            lhs.score == rhs.score &&
+            lhs.type == rhs.type
+    }
+}
+
+extension Podcast {
+    func getLastUpdatedAsDateWith(completion: @escaping (Date?) -> Void) {
+        DispatchQueue.global().async {
+            // slow calculations performed here
```

Comment:
What slow calculations do you anticipate performing here?

Reply:
let date = Date(iso8601String: self.date) is the slow "calculation" performed here. It's not super slow but the conversion does stop the main queue. Specifically if we want to use this function in a collection view.

Reply:
Wow, this is interesting. How did you catch this? Through the instrument tools? I learn something new.

Reply:
Yea I used instruments to identify every part of code that was getting the collectionViews under 60fps. That's why there is a ton of threading in this push.